packages/cli/src/commands/configure/configure_profile_command.ts

```diff
@@ -41,6 +45,12 @@ export class ConfigureProfileCommand
   handler = async (
     args: ArgumentsCamelCase<ConfigureProfileCommandOptions>
   ): Promise<void> => {
+    const usageDataEmitter = await this.usageDataEmitterCreator();
+    await usageDataEmitter.emitSuccess(
+      {},
+      { command: 'amplify configure profile' }
+    );
```

Comment:
Instead of doing this per command, see if we can use the `middleware` or `yargs.onFinishCommand` to capture contextual metrics data from the commands.

Also the failure metrics should be reported as well. You should be able to add it here https://github.com/aws-amplify/amplify-backend/blob/main/packages/cli/src/error_handler.ts#L44

Reply:
I was considering middleware for the approach, but it didn't seem like a good fit since those are trigger before the handler.

Reply:
I am not familiar with the yargs.onFinishCommand, do you have some documentation on this feature?

Reply:
It seems `onFinishCommand` was removed from the docs in favor of being able to await the parser, e.g. `await parser.parseAsync()`. That means you can emit metrics _after_ the parser finishes.

Reply:
Even the parseAsync seems to trigger after parsing CLI input, not after handler executes

Reply:
Can you use the return value of [this call](https://github.com/aws-amplify/amplify-backend/blob/dca2a0047c73ce3fae4023b29ec361512dc85635/packages/cli/src/amplify.ts#L11) to record success metrics in a generic way?